app/lib/katello/util/default_enablement_migrator.rb

```diff
@@ -0,0 +1,205 @@
+module Katello
+  module Util
+    class DefaultEnablementMigrator # used in lib/katello/tasks/upgrades/4.9/update_custom_products_enablement.rake
+      include ActionView::Helpers::TextHelper
+
+      def execute!
+        ds_errors = create_disabled_overrides_for_non_sca_org_hosts
+        dsak_errors = create_disabled_overrides_for_non_sca_org_activation_keys
+        ak_errors = create_activation_key_overrides
+        consumer_errors = create_consumer_overrides
+        cp_errors = update_enablement_in_candlepin
+        kt_errors = update_enablement_in_katello
+
+        total_errors = ds_errors + dsak_errors + ak_errors + consumer_errors + cp_errors + kt_errors
+        finish_message = "Finished updating custom products enablement; #{total_errors == 0 ? "no errors" : "#{pluralize(total_errors, "error")}"}"
+        Rails.logger.info finish_message
+        Rails.logger.info("#{pluralize(ds_errors, "error")} updating disabled overrides for unsubscribed content; see log messages above") if ds_errors > 0
+        Rails.logger.info("#{pluralize(dsak_errors, "error")} updating disabled overrides for unsubscribed content in activation keys; see log messages above") if dsak_errors > 0
+        Rails.logger.info("#{pluralize(ak_errors, "error")} updating activation key overrides; see log messages above") if ak_errors > 0
+        Rails.logger.info("#{pluralize(consumer_errors, "error")} updating consumer overrides; see log messages above") if consumer_errors > 0
+        Rails.logger.info("#{pluralize(cp_errors, "error")} updating default enablement in Candlepin; see log messages above") if cp_errors > 0
+        Rails.logger.info("#{pluralize(kt_errors, "error")} updating default enablement in Katello; see log messages above") if kt_errors > 0
+      end
+
+      # rubocop:disable Metrics/MethodLength
+      def create_disabled_overrides_for_non_sca_org_hosts
+        errors = 0
+        Organization.all.each do |org|
+          next if org.simple_content_access? # subscription attachment is meaningless with SCA
+          Rails.logger.info("Creating disabled overrides for unsubscribed content in org #{org.name}")
+          hosts_to_update = org.hosts.where.not(subscription_facet: nil)
+          hosts_to_update.each do |host|
+            content_finder = ::Katello::ProductContentFinder.new(
+              match_subscription: false,
+              match_environment: false,
+              consumable: host.subscription_facet
+            )
+            subscribed_content_finder = ::Katello::ProductContentFinder.new(
+              match_subscription: true,
+              match_environment: false,
+              consumable: host.subscription_facet
+            )
+            unsubscribed_content = content_finder.custom_content_labels - subscribed_content_finder.custom_content_labels
```

Comment:
Can you explain the logic here. I guess if both `match_subscription` and `match_environment` is false do we get all the  product contents for an org?